program_test.go

```diff
@@ -22,4 +22,15 @@ func TestProgram(t *testing.T) {
 	ss := p.Streams()
 	require.Equal(t, 1, len(ss))
 	require.Equal(t, s.ID(), ss[0].ID())
+
+	p.SetFlags(1)
+	require.Equal(t, 1, p.Flags())
+	p.SetProgramNum(101)
+	require.Equal(t, 101, p.ProgramNum())
+	p.SetDiscard(DiscardAll)
+	require.Equal(t, DiscardAll, p.Discard())
+	d := NewDictionary()
+	_ = d.Set("service_name", "test_service_name", 0)
```

Comment:
Can you wrap this inside a `require.NoError` instead? `require.NoError(t, d.Set("service_name", "test_service_name", 0))`